ocrd/ocrd/workspace.py

```diff
@@ -117,18 +119,31 @@ def download_file(self, f, _recursion_count=0):
             f.local_filename = f.url
             return f
 
-    def remove_file(self, ID, force=False, keep_file=False):
+    def remove_file(self, ID, force=False, keep_file=False, page_recursive=False, page_same_group=False):
         """
         Remove a file from the workspace.
 
         Arguments:
             ID (string|OcrdFile): ID of the file to delete or the file itself
             force (boolean): Continue removing even if file not found in METS
             keep_file (boolean): Whether to keep files on disk
+            page_recursive (boolean): Whether to remove all images referenced in the file if the file is a PAGE-XML document.
+            page_same_group (boolean): Remove only images in the same file group as the PAGE-XML. Has no effect unless ``page_recursive`` is ``True``.
         """
         log.debug('Deleting mets:file %s', ID)
+        if isinstance(ID, OcrdFile):
+            ID = ID.ID
         try:
             ocrd_file = self.mets.remove_file(ID)
+            if ocrd_file.mimetype == MIMETYPE_PAGE and page_recursive:
+                with pushd_popd(self.directory):
+                    ocrd_page = parse(self.download_file(ocrd_file).local_filename, silence=True)
+                    for img_url in ocrd_page.get_AllImagePaths():
+                        img_kwargs = {'url': img_url}
+                        if page_same_group:
+                            img_kwargs['fileGrp'] = ocrd_file.fileGrp
+                        for img_file in self.mets.find_files(**img_kwargs):
+                            self.remove_file(img_file, keep_file=keep_file, force=force)
```

Comment:
Very elegant!

Reply:
Wait, they way `get_AllImagePaths` is implemented (namely _both_ the original and all derived images), this would also remove the original image, right?

Reply:
Yes, should probably distinguish `page` into `page_filename` and `page_alternative`?

Reply:
Actually, I don't think that `/PcGts/Page/@imageFilename` needs to be there at all. The original image is never "owned" by any PAGE fileGrp in our functional model. And you never need to access it outside of `image_from_page`.
But if you want to keep it for other use cases, then perhaps an additional boolean kwarg `include_original=False` (or similar) would be more intuitive.

Reply:
Getting the `@imageFilename` is trivial, so I changed the method to only return `AlternativeImage/@filename` where the load-xml-and-xpath approach really makes sense. ab91268e